rest/middleware_test.go

```diff
@@ -1,238 +1,179 @@
-			}
-
-			if diff := cmp.Diff(&requestLog, tt.wantRequestLog, cmpopts.IgnoreFields(logMessage{}, "RequestID", "Body")); diff != "" {
-				t.Errorf("diff in entry log: %s", diff)
-			}
-
-			var closingLog logMessage
-			err = json.Unmarshal([]byte(lines[1]), &closingLog)
-			require.NoError(t, err)
-
-			assert.NotEmpty(t, closingLog.RequestID)
-			_, err = uuid.Parse(closingLog.RequestID)
-			require.NoError(t, err)
-
-			d, err := time.ParseDuration(closingLog.Duration)
-			require.NoError(t, err)
-			assert.Greater(t, int64(d), int64(0))
-
-			if tt.wantBody {
-				assert.NotEmpty(t, closingLog.Body)
-			}
-
-			if diff := cmp.Diff(&closingLog, tt.wantClosingLog, cmpopts.IgnoreFields(logMessage{}, "RequestID", "Duration", "Body")); diff != "" {
-				t.Errorf("diff in closing log: %s", diff)
-			}
-		})
-	}
-}
-
-type ZapTestLogger struct {
-	io.Writer
-
-	b       *bytes.Buffer
-	bwriter *bufio.Writer
-	logger  *zap.SugaredLogger
-}
-
-func (z ZapTestLogger) Close() error {
-	return nil
-}
-
-func (z ZapTestLogger) Sync() error {
-	return nil
-}
-
-func (z *ZapTestLogger) GetLogs() string {
-	z.bwriter.Flush()
-	return strings.TrimSpace(z.b.String())
-}
-
-func (z *ZapTestLogger) GetLogger() *zap.SugaredLogger {
-	return z.logger
-}
-
-func newZapTestLogger(t *testing.T, level zapcore.Level) *ZapTestLogger {
-	var b bytes.Buffer
-	bWriter := bufio.NewWriter(&b)
-
-	config := zap.NewProductionConfig()
-	config.EncoderConfig.FunctionKey = "function"
-
-	testLogger := ZapTestLogger{
-		Writer:  bWriter,
-		bwriter: bWriter,
-		b:       &b,
-	}
-
-	schemeName := strings.ReplaceAll(t.Name(), "/", "-")
-	schemeName = strings.ReplaceAll(schemeName, "_", "-")
-
-	err := zap.RegisterSink(schemeName, func(u *url.URL) (zap.Sink, error) {
-		return testLogger, nil
-	})
-	require.NoError(t, err)
-
-	customPath := fmt.Sprintf("%s:whatever", schemeName)
-	config.OutputPaths = []string{customPath}
-	config.Level = zap.NewAtomicLevelAt(level)
-	config.EncoderConfig.TimeKey = "timestamp"
-	config.EncoderConfig.EncodeTime = zapcore.RFC3339TimeEncoder
-
-	zapLogger, err := config.Build()
-	require.NoError(t, err)
-
-	testLogger.logger = zapLogger.Sugar()
-
-	return &testLogger
-}
-
-func sendRequestThroughFilterChain(t *testing.T, handler func(req *restful.Request, resp *restful.Response), filters ...restful.FilterFunction) {
-	ws := new(restful.WebService).Path("/").Consumes(restful.MIME_JSON).Produces(restful.MIME_JSON)
-
-	c := restful.NewContainer()
-	c.Add(ws.Route(ws.GET("test").To(handler)))
-	for _, f := range filters {
-		c.Filter(f)
-	}
-
-	httpRequest, err := http.NewRequestWithContext(context.TODO(), "GET", "http://localhost/test", nil)
-	require.NoError(t, err)
-	httpRequest.RemoteAddr = "1.2.3.4"
-	httpRequest.Header.Set("Accept", "application/json")
-
-	httpWriter := httptest.NewRecorder()
-
-	c.Dispatch(httpWriter, httpRequest)
-
-	require.Equal(t, http.StatusOK, httpWriter.Code)
-}
+// func TestRequestLoggerFilter(t *testing.T) {
```

Comment:
Is it planned to re-enable these tests? Do we need an issue?

Reply:
maybe a good Idea, would you please put these in a issue, including the test code. I will remove the code then here

Reply:
or we keep it commented, an add simple an issue. Actually i have no clue how i should migrate this test to slog

Reply:
Issue created: https://github.com/metal-stack/metal-lib/issues/130

Reply:
Without digging in, I am not sure either. But I assume we could implement a custom `slog.Handler` similar to the `ZapTestLogger`. I'd prefer to not include the commented code, but at least there should be a `TODO` eventually referencing the issue.